### PR TITLE
Framework: remove "AddNewButton" and replace with compact borderless Button

### PR DESCRIPTION
**Pull request by mtias**
The last few components using "AddNewButton" were in the editor. This PR gets rid of the dependency and the component.

image

### CODE REVIEW

File: client/post-editor/editor-sharing/publicize-options.jsx [hunk 1]
```diff
@@ -152,15 +152,12 @@ module.exports = React.createClass( {
...
-			<AddNewButton
-				isCompact={ true }
-				onClick={ this.newConnection }
-				>
-				{ this.translate( 'Connect new service' ) }
-			<span className="editor-sharing__external-link-indicator">
-				<Gridicon icon="external" size={ 18 } />
-			</span>
-			</AddNewButton>
+			<Button borderless compact={ true } onClick={ this.newConnection }>
```
Comment (aduth) [inline client/post-editor/editor-sharing/publicize-options.jsx]: Minor: A bit of mixed conventions here `borderless compact={ true }`

Should be either `borderless={ true } compact={true }` or `borderless compact`

My vote is for the latter :smile:

**Comment by aduth** [comment]
The "Connect new service" button lost some of the margin between itself and the message customization. Might give the impression that it's more related to the message than it is with the connections themselves.

Before|After
---|---
Before|After

**Comment by aduth** [comment]
I thought I'd hate the folder icon, but in practice it's grown on me.

A few cosmetic issues notwithstanding, this behaves well and I've confirmed no other usage of `<AddNewButton />`.

That said, there are a few lingering selectors for removed usages that should probably be cleaned up:
- https://github.com/Automattic/wp-calypso/blob/ca62f8a/client/my-sites/upgrades/domain-management/style.scss#L121-L124
- https://github.com/Automattic/wp-calypso/blob/ca62f8a/client/my-sites/plugins/plugin-site-jetpack/style.scss#L18-L20
- https://github.com/Automattic/wp-calypso/blob/ca62f8a/client/my-sites/plugins/plugin-site-network/style.scss#L17-L19

File: client/post-editor/editor-sharing/publicize-options.jsx (hunk 1 above)
Comment (mtias) [inline client/post-editor/editor-sharing/publicize-options.jsx]: Agreed.

**Pull request merged**